### ISSUE TITLE
some information about pretrain

### ISSUE THREAD

**Issue by tuotuo535** [jimmyrate/deepAMP#6]
Below is the code for the pretrain section. Is there anything written incorrectly?
such as, dataset='uniport_all_60',isPair=False, the peptides in the pretraining section are not paired, right?

sample_conf = makeSampleConfig(
    dataset='antifungal',isPair=True, isRandom=False, sample_times=50, batch_size=10, protein_path=text_path,
    mask=mask, mask_ratio=0.3, max_pred=20, num_workers=5,)

**Comment by jimmyrate** [comment]
You are right. In the pretraining stage, the datasets are not paired.

The sample_conf is used for the sample (generate new peptides), so the parameter （isPair=True）is correct in the config.

**Comment by tuotuo535** [comment]
> You are right. In the pretraining stage, the datasets are not paired.
> 
> The sample_conf is used for the sample (generate new peptides), so the parameter （isPair=True）is correct in the config.

Thank you for your reply, I understand. I'm sorry to bother you again, but I have a few more questions. 
1. What is the function of the do_sample.py file? Is it to load an existing model to generate new antimicrobial peptides? For example, in datapre.py, if dataset='uniport_all_60' is set, does it mean loading the pre-trained model to generate antimicrobial peptides? And if dataset='random_pair' is set, does it mean loading the model fine-tuned by AOM to generate antimicrobial peptides? Is my understanding correct?
2. When running the do_sample.py file, the fixedparameters.py file is used, in which SPECULATED_EPOCH = tuple(range(180+1,200+1)), that is, models from the 180th to the 199th epochs are utilized. Why are these values chosen?
I'm looking forward to your answers. Thank you.

**Comment by jimmyrate** [comment]
1. This function is about generated parameter control. Your understanding is correct; using dataset=‘uniport_all_60’, the pre-trained model is loaded. Specifically, you can debug to see the flow of the code.
2. Because the losses of the latter 20 models saved were minimized and the losses did not continue to decline, all we chose to use the latter 20 models ckpt to generate new antimicrobial peptides.

**Comment by tuotuo535** [comment]
> 1. This function is about generated parameter control. Your understanding is correct; using dataset=‘uniport_all_60’, the pre-trained model is loaded. Specifically, you can debug to see the flow of the code.
> 2. Because the losses of the latter 20 models saved were minimized and the losses did not continue to decline, all we chose to use the latter 20 models ckpt to generate new antimicrobial peptides.

Thank you very much!
